nixos/modules/programs/firefox/default.nix

```
@@ -0,0 +1,308 @@
+              "kn"
+              "ko"
+              "lij"
+              "lt"
+              "lv"
+              "mk"
+              "mr"
+              "ms"
+              "my"
+              "nb-NO"
+              "ne-NP"
+              "nl"
+              "nn-NO"
+              "oc"
+              "pa-IN"
+              "pl"
+              "pt-BR"
+              "pt-PT"
+              "rm"
+              "ro"
+              "ru"
+              "sat"
+              "sc"
+              "sco"
+              "si"
+              "sk"
+              "skr"
+              "sl"
+              "son"
+              "sq"
+              "sr"
+              "sv-SE"
+              "szl"
+              "ta"
+              "te"
+              "tg"
+              "th"
+              "tl"
+              "tr"
+              "trs"
+              "uk"
+              "ur"
+              "uz"
+              "vi"
+              "xh"
+              "zh-CN"
+              "zh-TW"
+            ]
+          );
+          default = [ ];
+          description = ''
+            The language packs to install.
+          '';
+        };
+
+        autoConfig = lib.mkOption {
+          type = lib.types.lines;
+          default = "";
+          description = ''
+            AutoConfig files can be used to set and lock preferences that are not covered
+            by the policies.json for Mac and Linux. This method can be used to automatically
+            change user preferences or prevent the end user from modifiying specific
+            preferences by locking them. More info can be found in <https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig>.
+          '';
+        };
+
+        autoConfigFiles = lib.mkOption {
+          type = with lib.types; listOf path;
+          default = [ ];
+          description = ''
+            AutoConfig files can be used to set and lock preferences that are not covered
+            by the policies.json for Mac and Linux. This method can be used to automatically
+            change user preferences or prevent the end user from modifiying specific
+            preferences by locking them. More info can be found in <https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig>.
+
+            Files are concated and autoConfig is appended.
+          '';
+        };
+
+        nativeMessagingHosts.packages = lib.mkOption {
+          type = lib.types.listOf lib.types.package;
+          default = [ ];
+          description = ''
+            Additional packages containing native messaging hosts that should be made available to ${prettyName} extensions.
+          '';
+        };
+      };
+
+      config = lib.mkIf cfg.enable {
+        environment.systemPackages = [
+          (cfg.package.override (old: {
+            extraPrefsFiles =
+              (old.extraPrefsFiles or [ ])
+              ++ cfg.autoConfigFiles
+              ++ [ (pkgs.writeText "autoconfig.js" cfg.autoConfig) ];
```

```suggestion
              ++ [ (pkgs.writeText "${variant}-autoconfig.js" cfg.autoConfig) ];
```
I still feel like this should include the name, with `firefox-autoconfig.js` it's easy to tell at a glance what the file is while just `autoconfig.js` could be anything.